tests/common/mod.rs

```diff
@@ -0,0 +1,188 @@
+        "-zmqpubrawblock=tcp://127.0.0.1:28332",
+        "-zmqpubrawtx=tcp://127.0.0.1:28333",
+    ];
+    let bitcoind = BitcoinD::from_downloaded_with_conf(&conf).unwrap();
+
+    // Mine 101 blocks in our little regtest network so that the funds are spendable.
+    // (See https://bitcoin.stackexchange.com/questions/1991/what-is-the-block-maturation-time)
+    let address = bitcoind
+        .client
+        .get_new_address(None, Some(json::AddressType::Bech32))
+        .unwrap();
+    let address = address.require_network(Network::Regtest).unwrap();
+    bitcoind.client.generate_to_address(101, &address).unwrap();
+
+    (bitcoind, bitcoind_dir)
+}
+
+#[cfg(all(target_os = "linux", target_arch = "x86_64"))]
+fn lnd_download_filename() -> String {
+    format!("lnd-linux-amd64-v{}-beta.tar.gz", &LND_VERSION)
+}
+
+// download_lnd downloads the lnd binary to tmp if it doesn't exist yet. Currently it only downloads binaries
+// compatible with the linux os.
+pub async fn download_lnd() -> TempDir {
+    let temp_dir = tempdir().unwrap();
+    let lnd_dir = &temp_dir.path().to_path_buf();
+    let lnd_releases_endpoint =
+        "https://github.com/lightningnetwork/lnd/releases/download/v0.16.0-beta";
+    let lnd_download_endpoint = format!("{lnd_releases_endpoint}/{}", lnd_download_filename());
+
+    let resp = minreq::get(&lnd_download_endpoint).send().unwrap();
+    assert_eq!(
+        resp.status_code, 200,
+        "url {lnd_download_endpoint} didn't return 200"
+    );
+    let content = resp.as_bytes();
+
+    let decoder = GzDecoder::new(&content[..]);
+    let mut archive = Archive::new(decoder);
+    archive.unpack(lnd_dir).expect("unable to unpack lnd files");
+
+    temp_dir
+}
+
+// LndNode holds the tools we need to interact with a Lightning node.
+#[derive(Debug)]
+pub struct LndNode {
+    address: String,
+    _lnd_exe_dir: TempDir,
+    _lnd_dir_tmp: TempDir,
+    cert_path: String,
+    _handle: Child,
+}
+
+impl LndNode {
+    fn new(bitcoind_connect_params: ConnectParams, lnd_exe_dir: TempDir) -> LndNode {
+        env::set_current_dir(lnd_exe_dir.path().join("lnd-linux-amd64-v0.16.0-beta"))
+            .expect("couldn't set current directory");
+
+        let lnd_dir_binding = Builder::new().prefix("lnd").tempdir().unwrap();
+        let lnd_dir = lnd_dir_binding.path();
+
+        let cert_path = lnd_dir.join("tls.cert").to_str().unwrap().to_string();
+
+        // Have node run on a randomly assigned grpc port. That way, if we run more than one lnd node, they won't
+        // clash.
+        let port = bitcoind::get_available_port().unwrap();
+        let rpc_addr = format!("localhost:{}", port);
+        let args = [
+            format!("--rpclisten={}", rpc_addr),
+            format!("--norest"),
+            format!("--bitcoin.active"),
+            format!("--bitcoin.node=bitcoind"),
+            format!("--bitcoin.regtest"),
+            format!("--lnddir={}", lnd_dir.display()),
```

Comment:
It would be very useful to have the logs from LND +  LNDK saved by itests under a directory with that test's name so that you can debug CI failures, possibly set log dir to something more permanent.